packages/components/src/components/modal/_modal.scss

```diff
@@ -83,31 +84,98 @@
     transition: transform $duration--moderate-02 motion(exit, expressive);
 
     @include carbon--breakpoint(md) {
-      width: 50%;
+      width: 84%;
       max-width: 768px;
       max-height: 90%;
       height: auto;
     }
 
     @include carbon--breakpoint(lg) {
-      max-height: 80%;
+      width: 60%;
+      max-height: 84%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 48%;
+    }
+  }
+
+  .#{$prefix}--modal-container--xs {
+    @include carbon--breakpoint(md) {
+      width: 48%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 32%;
+      max-height: 48%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 24%;
+
+      .#{$prefix}--modal-header,
+      .#{$prefix}--modal-content,
+      .#{$prefix}--modal-content__regular-content {
+        padding-right: rem(16px);
```

Comment:
should we use a spacing token here instead?

Reply:
Please see: https://github.com/carbon-design-system/carbon/pull/4657#discussion_r345546598

Still a good catch, though, as the code I referred to doesn't use `rem()` function. Rectified at: https://github.com/carbon-design-system/carbon/pull/4657/commits/164817ea3732df2982664f106bce752260a64961